syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java

```diff
@@ -209,6 +277,11 @@ public abstract static class SyndeoPubsubReadSchemaTransformConfiguration {
             + "For JSON data, this is a schema defined with JSON-schema syntax (https://json-schema.org/).")
     public abstract String getSchema();
 
+    // Used for testing.
+    public abstract @Nullable PubsubTestClientFactory getClientFactory();
+
+    public abstract @Nullable Clock getClock();
+
```

Comment:
please add a description stating that these fields should only be used for testing (see the fields above)